### PR TITLE
chore: Extract concatMap body

### PR DESCRIPTION
**Pull request by ukstv**

### CODE REVIEW

File: packages/core/src/state-management/repository-internals.ts [hunk 1]
```diff
@@ -294,6 +294,78 @@ export class RepositoryInternals {
...
   }
 
+  /**
+   * Handle CASResponse and update state$.
+   *
+   * @param state$ - RunningState instance to update.
+   * @param asr - response from CAS.
+   * @return boolean - `true` if polling should stop, `false` if polling continues
+   */
+  async handleAnchorResponse(state$: RunningState, asr: CASResponse): Promise<boolean> {
```
Comment (dbcfd) [inline packages/core/src/state-management/repository-internals.ts:304]: Why asr as the name for the cas response? Can we name it `casResponse` or just `resp`?

**Review by dbcfd — CHANGES_REQUESTED** [review]

**Review by stephhuynh18 — APPROVED** [review]

File: packages/core/src/state-management/repository-internals.ts (hunk 1 above)
Comment (ukstv) [reply packages/core/src/state-management/repository-internals.ts:304]: gonna address that in a followup PR

Comment (ukstv) [reply packages/core/src/state-management/repository-internals.ts:304]: or, okay

Comment (ukstv) [reply packages/core/src/state-management/repository-internals.ts]: renamed

**Review by dbcfd — APPROVED** [review]

**Pull request merged**